File: backend/src/main/java/com/lms_erp/employee/service/DesignationService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.DesignationRequest;
import com.lms_erp.employee.dto.DesignationResponse;

import java.util.List;

public interface DesignationService {

    DesignationResponse createDesignation(DesignationRequest request);

    DesignationResponse updateDesignation(Long designationId,
                                          DesignationRequest request);

    DesignationResponse getDesignationById(Long designationId);

    List<DesignationResponse> getAllDesignations();

    void deleteDesignation(Long designationId);
}

File: lms-frontend/src/employee/services/designationService.js
// designationService.js

import api from "../../api/axios";

const BASE_URL = "/api/designations";

const getAllDesignations = () => {

    return api.get(
        BASE_URL
    );
};

const getDesignationById = (designationId) => {

    return api.get(
        `${BASE_URL}/${designationId}`
    );
};

const createDesignation = (designationData) => {

    return api.post(
        BASE_URL,
        designationData
    );
};

const updateDesignation = (
    designationId,
    designationData
) => {

    return api.put(
        `${BASE_URL}/${designationId}`,
        designationData
    );
};

const deleteDesignation = (designationId) => {

    return api.delete(
        `${BASE_URL}/${designationId}`
    );
};

const designationService = {

    getAllDesignations,
    getDesignationById,
    createDesignation,
    updateDesignation,
    deleteDesignation
};

export default designationService;